sdk/python/kfp/dsl/types.py

```diff
@@ -94,6 +98,35 @@ class InconsistentTypeException(Exception):
 	'''InconsistencyTypeException is raised when two types are not consistent'''
 	pass
 
+
+class InconsistentTypeWarning(Warning):
+	'''InconsistentTypeWarning is issued when two types are not consistent'''
+	pass
+
+
+def verify_type_compatibility(given_type, expected_type, error_message_prefix : str = ''):
+	'''verify_type_compatibility verifies that the given argument type is compatible with the expected input type.
+	Args:
+		given_type (str/dict): The type of the argument passed to the input
+		expected_type (str/dict): The declared type of the input
+	'''
+	if given_type is None:
```

Comment:
Is that okay to combine these two 'if'?

Reply:
Done

Reply:
Sorry I am not familiar with this enough. I am curious when these two params can be none and why no error/warning should be thrown.

Reply:
We support checking type compatibility as a convenience for the pipeline author.
Currently we do not require the inputs and outputs to by typed an they often do not have types specified. We consider that by not providing a type, the user opts-out of the type checking convenience.